src/main/scala/is/hail/io/bgen/BgenRecord.scala

```diff
@@ -150,13 +150,13 @@ class BgenRecordV12(compressed: Boolean, nSamples: Int, tolerance: Double) exten
           while (i < nGenotypes - 1) {
             assert(probabilityIterator.hasNext, "Did not decode bytes correctly. Ran out of probabilities.")
             val p = probabilityIterator.next()
-            sampleProbs(i) = p
+            sampleProbs.update(i, p)
```

Comment:
You don't need to change this, `a(i) = x` is syntactic sugar for `a.update(i, x)`, like `apply`.